server/sonar-server/src/main/java/org/sonar/server/computation/period/Period.java

```diff
@@ -22,19 +22,22 @@
 
 import javax.annotation.CheckForNull;
 import javax.annotation.Nullable;
-import org.apache.commons.lang.builder.ReflectionToStringBuilder;
+import javax.annotation.concurrent.Immutable;
 
+@Immutable
 public class Period {
```

Comment:
I've got a cleaner version of this class in PR https://github.com/SonarSource/sonarqube/pull/384, you should rebase on it to add this snapshotId property